### DISCUSSION TITLE
After every config change in Loxone, I need to restart HA in order for the plugin to work

### DISCUSSION THREAD

**Discussion by Rubeast** [JoDehli/PyLoxone#315] [Q&A]
I don't know if I'm the only one with this "problem". But every time I make a change in the Loxone config, the PyLoxone plugin stops working. All switches seem to get frozen in the state they were at the moment of the Loxone change and when I toggle switches via HA, they don't seem to communicate with Loxone.

Are other people also experiencing this? For me it is not really a problem, but I just often forget to do the HA restart and only notice it after a while when I notice some things are no longer working.
I was thinking about a way to automate a HA restart after every Loxone config change, but not really sure what the best way to implement this is.

Any thoughts?

**Comment by JoDehli** [comment]
@Rubeast this is indeed a problem but at the moment it will not be fixed. We have to accept it for the moment.

**Reply by Rubeast** [reply]
Good to know it is not a problem just for me. But if there is no mechanism to check if the connection to Loxone still works I will try to make something myself. I was thinking about a switch in Loxone that isn't connected to something but would toggle each 30sec. Then have an automation in HA check if the toggle has been switched in the last minute or two and have it take action when that isn't the case. (a notification or maybe even a restart of HA)

**Reply by JoDehli** [reply]
@Rubeast this is a bad idea. This could cause connections errors as well. The connection should only lost if the loxone mini server is restarted and normally you know when you restart it. My connection is stable over weeks and month if I do not restart the mini server.

**Reply by Rubeast** [reply]
Ok thx for the info. I will try with just a notification first. Can't do anything wrong with that I think?

I know it is something simple and not really an annoyance for most people. 
But it is something I forget almost every time and only notice after a while when I notice that something isn't working. I need a way to remind myself 😁

**Reply by JoDehli** [reply]
@Rubeast as I nearly never change something in my loxone I have no problems. On my machine also the reconnect after the loxone config is successful. Maybe your system is so big that the restart takes so long. 

I work also on a new implementation but which should make the system more robust. I have only problems with very big Systems
